build.py

```diff
@@ -154,7 +154,7 @@ def install_llvm_tool(name, source_location, prefix_root, debug, compiler, jobs,
     if signature == disk_signature:
         print('LLVM:{} + Binder install is detected at {}, skipping LLVM installation and Binder building procedures...\n'.format(llvm_version, build_dir))
 
-    else:
+    elif not update_binder:
```

Comment:
This seems a bit odd, - if signature does not match then it is clear that Binder upgrade is needed. Could you please elaborate on the logic here? My initial impression for a new `--update-binder` option would be that it will trigger script to check if such upgrade is needed and rebuild Binder if so. Do you see this differently? If so please elaborate.

Reply:
Hm… it sounds like there is a disconnect here, please let me know if below is not what you are observing on your system:
-- there should not be automatic rebuild unless binder repository have a new commits, - ie changing code should not trigger the rebuilds.
-- to avoid re-build checks (and potential rebuilds) the simplest way is to pass `--binder <path-to-binary>` option when using `build-and-run-tests.py`.

Note that I do like the idea of having `--update-binder` option, but I though it will trigger be something like "rebuild binder binaries even if no new commits are found" so its implementation should probably just call `ninja` in the right dir (this is what usually do by hands)